src/typing/flow_js.ml

```diff
@@ -4444,17 +4444,17 @@ let rec __flow cx ((l: Type.t), (u: Type.use_t)) trace =
         else SMap.remove "constructor" proto_props
       in
       let props = SMap.union own_props proto_props in
-      match_shape cx trace ~use_op proto reason props
+      match_shape cx trace ~use_op proto reason reason_op props
 
     (* Function definitions are incompatible with ShapeT. ShapeT is meant to
      * match an object type with a subset of the props in the type being
      * destructured. It would be complicated and confusing to use a function for
      * this.
      *
      * This invariant is important for the React setState() type definition. *)
-    | _, UseT (use_op, ShapeT o) ->
+    | _, UseT (use_op, ShapeT (reason_op, o)) ->
       add_output cx ~trace
-        (Error_message.EIncompatibleWithShape (reason_of_t l, reason_of_t o, use_op))
+        (Error_message.EIncompatibleWithShape (reason_op, reason_of_t o, use_op))
```

Comment:
i think this should probably be `(reason_of_t l, reason_op, use_op)`

Reply:
done